sdks/java/core/build.gradle

```diff
@@ -117,6 +132,113 @@ task packageTests(type: Jar) {
   classifier = "tests"
 }
 
+class ValidatesRunnerConfig {
+  // Task name prefix
+  String testPrefix
+  // Runner name in documentation string
+  String runnerName
+  // List of test categories to exclude from this task. Optional.
+  List<String> excludes
+  // Pipeline options command line arguments.
+  // TODO: Can option keys be specified multiple times?
+  Map<String, String> pipelineOptions
+  // Configuration to use for test runtime classpath.
+  FileCollection configuration
+  // Additional system properties to be set for tests. Optional.
+  Map<String, String> systemProperties
+}
+
+def createValidatesRunner(Map m) {
```

Comment:
Use the implicit `it` parameter which allows you to call this method with zero or more args, see:
https://github.com/apache/beam/blob/master/build_rules.gradle#L164

I like the method but this could be built as a for loop over a set of maps and known runners/configurations like in examples/java. Using the for loop attempts to adhere that as few specializations per runner are added.

Reply:
This seems like it harms readability for people unfamiliar with groovy. I also find this more readable than the examples build file because configurations are explicitly linked by `ValidatesRunnerConfig` rather than name substitution. If nothing else, this at least makes it less likely that task/variable names collide unintentionally.

I do like the idea of minimizing specializations for runners, but at the end of the day, different runners have different configurations: classpaths, excluded categories, and even forking strategies. Does using a bunch of string maps really help?

Reply:
How about putting all configs into a list/map and iterating over that in a loop?